_episodes/41-code-review.md

```diff
@@ -213,281 +153,393 @@ you will need to specify the base branch and the head branch.
+   and then submit the pull request.
+5. Repository moderator and other collaborators on the repository (code reviewers)
+   will be notified of your pull request by GitHub.
+6. At this point, the code review process is initiated.
+
+Find someone else to swap repositories with (i.e. you will review the pull request
+they just raised on their repository). If there are an odd number, three people could
+go round robin (i.e. person A reviews code on person Bs repo, who reviews code on
+person C's repository, who reviews the code on person A's repository).
+
+If you are completing the course on your own, you will switch to being the reviewer
+of the pull request you just opened.
+
+### Things to look for in a code review
+
+Reviewing code effectively takes practice.
+However, here is some guidance on the kinds of things you should
+be looking for when reviewing a piece of code.
+
+Start by understanding what the code _should_ do, by reading the specification/user requirements,
+the pull request description and talking to the developer.
+In this case, understand what SR1.1.1 means.
+
+Once you're happy, start reading the code (skip the test code for now). You're
+going to be assessing the code in 4 key areas:
+
+* Is the code readable
+* Is the code a minimal change
+* Is the structure of the code clear
+* Is there appropriate and up-to-date documentation
+
+#### Is the code readable
+
+Think about do the names of the variables, do they [follow guidelines for good
+names?](../15-coding-conventions/index.html#l#naming-conventions)
+
+Do you understand what conditions in each if statements are for?
+
+Do the function names match the behavior of the function.
+
+#### Is the code a minimal change
+
+Does the code reimplement anything that already exists, either
+elsewhere in the codebase or in a library you know about?
+
+Does the code implement something that isn't on the ticket?
+
+#### Is the structure of the code clear
+
+Do functions do just one thing? Have appropriate design
+patterns been used (e.g. separating out the model logic from
+any view considerations)?
+
+#### Is there appropriate and up-to-date documentation
+
+If functionality has changed, has corresponding documentation been
+updated? If new functions have been added, do they have appropriate
+levels of documentation? Does the documentation make sense?
+
+Are there clear and useful comments that explain complex designs
+and focus on the "why/because" rather than the "what/how"?
+
+### Effective review comments
+
+Make sure your review comments are specific and actionable.
+
+Try to be as specific as you can, rather than "this code is unclear"
+prefer, "I don't understand what values this variable can hold".
+
+Make it clear in the comment if you want something to change as part
+of this PR. Ideally provide a concrete suggestion (e.g. better variable name).
+
+> ## Exercise: review some code
+>
+> Pick someone else in the group and go to the pull request they created on their repo.
```

Comment:
Explicitly show before hand how to add comments, and make explicit they should be added to the review but not submitted yet